nix/packages/docs-backend.nix

```diff
@@ -0,0 +1,124 @@
+{
+  lib,
+  python3,
+  fetchpatch,
+  fetchFromGitHub,
+}:
+let
+  python = python3.override {
+    self = python3;
+    packageOverrides = self: super: {
+      django = super.django_5;
+
+      django-extensions = super.django-extensions.overridePythonAttrs (oldAttrs: {
+        nativeCheckInputs = [ self.pyasyncore ];
+
+        patches = lib.optional (lib.versionOlder oldAttrs.version "4.0") (fetchpatch {
+          url = "https://patch-diff.githubusercontent.com/raw/browniebroke/django-extensions/pull/2.patch";
+          hash = "sha256-oYRfchotvv7E4xOuicPEVjfRZ2jiKfTPHWQZ/+YLE2g=";
+        });
+      });
+    };
+  };
+in
+
+python.pkgs.buildPythonApplication rec {
+  pname = "docs-backend";
+  version = "3.1.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "suitenumerique";
+    repo = "docs";
+    tag = "v${version}";
+    hash = "sha256-QeHtQWeCk1jwwtbPgQVtTKGIOUnsg9uERgrYNDl8xRk=";
+  };
+
+  sourceRoot = "source/src/backend";
+
+  build-system = with python.pkgs; [ setuptools ];
+
+  dependencies = with python.pkgs; [
```

Comment:
Does that means we have to maintain this list of dependencies in addition to the already existing one (`pyproject.toml` file) ?

Reply:
If it's too much of a maintenance churn, the Nix packaging could be relegated to a "best effort" support level and Nix maintainers could be pinged to help on every PR that adds / removes a new package (hopefully, this doesn't happen too regularly?!).

Reply:
And it's not possible to parse or use already existing python tools to maintain this list ? It's not error prone to try to maintain the same list in different places ?
I feel we have to do the same job twice between this file and the `pyproject.toml` one.

Reply:
There's projects to do so in https://pyproject-nix.github.io/ but the Nix list and the pyproject.toml list does not do the same thing.

pyproject.toml only care about direct Python dependencies, wheras, the Nix list cares about the exhaustive list.

Furthermore, the Nix list relies on the security maintenance of nixpkgs whereas pyproject.toml relies on the correctness of the version markers.

I think moving to `pyproject.nix` can be done later on slowly as Docs is upstreamed in nixpkgs and then the Docs repo can be advertised to provide packaging for the bleeding edge of Docs while advertising that the package provenance is Docs' responsibility, whereas Nixpkgs would offer the Docs with package provenance from Nixpkgs. It's a classical pattern but it takes a bit of time to get right.

Reply:
Thanks for the explanation.
If the package will be in nixpkgs, why this PR is needed in docs repo ? Maybe I'm wrong but are you not doing the same job twice ?

Reply:
So the idea here is more to have a dev stack manage with Nix ?

Reply:
If it's the idea then to have less work I think we should test in the CI that the Nix stack is always working as expected. We can assert that it is "building" and then the project running. WDYT ?